### ISSUE TITLE
Cell types not separated in CellSpace

### ISSUE THREAD

**Issue by WhaleGe** [zakieh-tayyebi/CellSpace#2]
Hello, Thank you very much for the tool you developed. But I have some problems, **why can't my cell types be separated**. The sample have 7879 cells

![image](https://user-images.githubusercontent.com/32753674/177074289-cb42581d-24d7-4198-8a8a-d60816a739df.png)
![image](https://user-images.githubusercontent.com/32753674/177074351-03153656-d290-438c-a078-1db440677390.png)

My code is there：
```bash
CellSpace -output ./CellSpace_out/P11_T_0_embedding.tsv -cpMat ./CellSpace_out/P11_T_0_cell_by_tile_bin.mtx -peaks ./CellSpace_out/P11_T_0_top50K_variable_tiles.fa -sampleLen 150 -ngrams 3 -exmpPerPeak 20 -epoch 20
```

**Comment by zakieh-tayyebi** [comment]
> Hello, Thank you very much for the tool you developed. But I have some problems, why can't my cell types be separated.

Hi!

Is it possible that the order of the cells is incorrect? It seems like the cell types are randomly scattered which is unlikely. In my experience, even if CellSpace isn't working well, it would recover some structure.

I've explained [here](https://github.com/zakieh-tayyebi/CellSpace/issues/1#issuecomment-1158182494) that the cells may have a different order in the output matrix, and the R function that creates the CellSpace object should fix that.
Basically, the label index (for example __label__C1) has the same index as in the input count matrix.

**Comment by WhaleGe** [comment]
> Hi!
> 
> Is it possible that the order of the cells is incorrect? It seems like the cell types are randomly scattered which is unlikely. In my experience, even if CellSpace isn't working well, it would recover some structure.
> 
> I've explained [here](https://github.com/zakieh-tayyebi/CellSpace/issues/1#issuecomment-1158182494) that the cells may have a different order in the output matrix, and the R function that creates the CellSpace object should fix that. Basically, the label index (for example __label__C1) has the same index as in the input count matrix.

I'm so sorry to bother you again. I followed your preprocessing steps exactly from the bam file, but in ArchR's umap graph the cell types are **separable**. And it seems that when constructing the CellSpace object, the label can be automatically corresponding (for example: __label__C1 corresponds to the first row name of cellColData in ArchR). I checked the label correspondence several times, but still did not solve the problem. 
```R
cso <- CellSpace(
  project = id,
  emb.file = paste0(indir,"CellSpace_out/",id,"_embedding.tsv"), # cell and k-mer embeddings
  meta.data = cell.md[, c("Celltype", "Sample", "ArchR.cluster")],
  cell.names = rownames(cell.md)
)
```
![image](https://user-images.githubusercontent.com/32753674/178446169-052a0137-d4b2-4445-9d03-a5ccf1f10c53.png)

**Comment by zakieh-tayyebi** [comment]
> > Hi!
> > Is it possible that the order of the cells is incorrect? It seems like the cell types are randomly scattered which is unlikely. In my experience, even if CellSpace isn't working well, it would recover some structure.
> > I've explained [here](https://github.com/zakieh-tayyebi/CellSpace/issues/1#issuecomment-1158182494) that the cells may have a different order in the output matrix, and the R function that creates the CellSpace object should fix that. Basically, the label index (for example __label__C1) has the same index as in the input count matrix.
> 
> I'm so sorry to bother you again. I followed your preprocessing steps exactly from the bam file, but in ArchR's umap graph the cell types are **separable**. And it seems that when constructing the CellSpace object, the label can be automatically corresponding (for example: __label__C1 corresponds to the first row name of cellColData in ArchR). I checked the label correspondence several times, but still did not solve the problem.
> 
> ```r
> cso <- CellSpace(
>   project = id,
>   emb.file = paste0(indir,"CellSpace_out/",id,"_embedding.tsv"), # cell and k-mer embeddings
>   meta.data = cell.md[, c("Celltype", "Sample", "ArchR.cluster")],
>   cell.names = rownames(cell.md)
> )
> ```
> 
> ![image](https://user-images.githubusercontent.com/32753674/178446169-052a0137-d4b2-4445-9d03-a5ccf1f10c53.png)

Hi!
I know this is a very old issue, and the GitHub repo was not being maintained for a while, but I was wondering if there are any updates.
Did you every get a better result or have any idea what the issue was?
